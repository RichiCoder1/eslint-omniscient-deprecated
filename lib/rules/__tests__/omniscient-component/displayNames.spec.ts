import { fix } from "./__setup__/setup";

describe("omniscient fixer", () => {
    it("should raw no display name", () => {
        const input = `
import component from 'omniscient';
const Test = component(() => {})`;
        const report = fix(input);
        expect(report).toBeFixed();
        expect(report).toMatchSnapshot();
    });

    it("should string display name", () => {
        const input = `
import component from 'omniscient';
const Test = component('a', () => {})`;
        const report = fix(input);
        expect(report).toBeFixed();
        expect(report).toMatchSnapshot();
    });

    it("should template display name", () => {
        const input = `
import component from 'omniscient';
let b = "b";
const Test = component(\`a\${b}\`, () => {})`;
        const report = fix(input);
        expect(report).toBeFixed();
        expect(report).toMatchSnapshot();
    });

    it("should binary display name", () => {
        const input = `
import component from 'omniscient';
let b = "b";
const Test = component('a' + b, () => {})`;
        const report = fix(input);
        expect(report).toBeFixed();
        expect(report).toMatchSnapshot();
    });
});
