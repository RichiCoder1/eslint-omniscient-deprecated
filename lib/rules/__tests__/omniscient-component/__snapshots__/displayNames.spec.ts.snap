// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`omniscient fixer should binary display name 1`] = `
Object {
  "fixed": true,
  "messages": Array [],
  "output": "import { Component } from 'react';
import { isEqualImmutable } from 'ImmutableComponent';

import component from 'omniscient';
let b = \\"b\\";
class Test extends Component {
    static displayName = \\"a\\" + b;

    shouldComponentUpdate(nextProps, nextState) {
        return !(
            isEqualImmutable(this.props, nextProps) &&
            ((!oldState && !newState) ||
                isEqualImmutable(this.state, nextState))
        );
    }

    render() {}
}",
}
`;

exports[`omniscient fixer should raw no display name 1`] = `
Object {
  "fixed": true,
  "messages": Array [],
  "output": "import { Component } from 'react';
import { isEqualImmutable } from 'ImmutableComponent';

import component from 'omniscient';
class Test extends Component {
    shouldComponentUpdate(nextProps, nextState) {
        return !(
            isEqualImmutable(this.props, nextProps) &&
            ((!oldState && !newState) ||
                isEqualImmutable(this.state, nextState))
        );
    }

    render() {}
}",
}
`;

exports[`omniscient fixer should string display name 1`] = `
Object {
  "fixed": true,
  "messages": Array [],
  "output": "import { Component } from 'react';
import { isEqualImmutable } from 'ImmutableComponent';

import component from 'omniscient';
class Test extends Component {
    static displayName = \\"a\\";

    shouldComponentUpdate(nextProps, nextState) {
        return !(
            isEqualImmutable(this.props, nextProps) &&
            ((!oldState && !newState) ||
                isEqualImmutable(this.state, nextState))
        );
    }

    render() {}
}",
}
`;

exports[`omniscient fixer should template display name 1`] = `
Object {
  "fixed": true,
  "messages": Array [],
  "output": "import { Component } from 'react';
import { isEqualImmutable } from 'ImmutableComponent';

import component from 'omniscient';
let b = \\"b\\";
class Test extends Component {
    static displayName = \`a\${b}\`;

    shouldComponentUpdate(nextProps, nextState) {
        return !(
            isEqualImmutable(this.props, nextProps) &&
            ((!oldState && !newState) ||
                isEqualImmutable(this.state, nextState))
        );
    }

    render() {}
}",
}
`;
